opba-protocols/hbci-protocol-tests/hbci-bdd-sandbox/src/test/java/de/adorsys/opba/protocol/hbci/tests/e2e/sandbox/HbciSandboxConsentE2EHbciProtocolTest.java

```diff
@@ -236,13 +244,51 @@ void testDirectTransactionListWithoutSca() {
                 );
     }
 
+    @Test
+    void testAccountsListCacheUpdate() {
+        testAccountsListWithConsentNoScaButUserHasOneSca();
+        addAccount();
+        // get accounts without cache update and see that no new account in the result
+        then().open_banking_can_read_anton_brueckner_hbci_account_data_using_consent_bound_to_service_session_bank_blz_30000003();
+        // get accounts with cache update and see new account in the result
+        getUpdatedListAccounts();
+    }
+
+    private void addAccount() {
+        Account newAccount = new Account();
+        newAccount.setNumber("3333333333");
+        newAccount.setBalance(BigDecimal.valueOf(1000));
+        newAccount.setCurrency(Currency.getInstance("EUR"));
+
+        User user = config.getUsers().get(0);
+        user.getAccounts().add(newAccount);
+    }
+
+    private void getUpdatedListAccounts() {
+        when()
+                .fintech_calls_list_accounts_for_anton_brueckner_with_cache_update(BANK_BLZ_30000003_ID)
+                .and()
+                .user_logged_in_into_opba_as_opba_user_with_credentials_using_fintech_supplied_url(OPBA_LOGIN, OPBA_PASSWORD)
+                .and()
+                .user_anton_brueckner_provided_initial_parameters_to_list_accounts_with_all_accounts_consent()
+                .and()
+                .user_anton_brueckner_provided_correct_pin_to_embedded_authorization_and_sees_redirect_to_fintech_ok();
+        then()
+                .open_banking_has_consent_for_anton_brueckner_account_list()
+                .fintech_calls_consent_activation_for_current_authorization_id()
+                .open_banking_can_read_anton_brueckner_new_hbci_account();
+    }
+
+
+    @Test
+    void testTransactionListCacheUpdate() {
+        testDirectTransactionListWithSca();
+    }
+
     private void makeHbciAdapterToPointToHbciMockEndpoints() {
         adapterProperties.getAdorsysMockBanksBlz().stream()
                 .flatMap(it -> bankProfileJpaRepository.findByBankBankCode(String.valueOf(it)).stream())
-                .map(it -> {
-                    it.setUrl("http://localhost:" + port + "/hbci-mock/");
-                    return it;
-                })
+                .peek(it -> it.setUrl("http://localhost:" + port + "/hbci-mock/"))
```

Comment:
Avoid using `peek` to modify the content of the stream,
https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#peek-java.util.function.Consumer-

>API Note:
    This method exists mainly to support debugging, where you want to see the elements as they flow past a certain point in a pipeline: